chaosaws/asg/probes.py

```diff
@@ -59,6 +63,161 @@ def desired_equals_healthy_tags(tags: List[Dict[str, str]],
+                                         secrets: Secrets = None
+                                         ) -> AWSResponse:
+    """
+    Wait until desired number doesn't match the number of healthy instances
+
+    for each of the auto-scaling groups matching tags provided
+
+    `tags` are  expected as:
+    [{
+        'Key': 'KeyName',
+        'Value': 'KeyValue'
+    },
+    ...
+    ]
+
+    Returns: Integer (number of seconds it took to wait)
+    or sys.maxsize in case of timeout
+    """
+
+    if not tags:
+        raise FailedActivity(
+            "Non-empty tags is required")
+
+    client = aws_client('autoscaling', configuration, secrets)
+
+    start = time.time()
+
+    while True:
+        groups_descr = get_asg_by_tags(tags, client)
+        logger.debug(groups_descr)
+        result = is_desired_equals_healthy(groups_descr)
+        time.sleep(0.1)
+
+        if (time.time() - start) > timeout:
+            return maxsize
+
+        if not result:
+            return int(time.time() - start)
+
+
+def wait_desired_equals_healthy_tags(tags: List[Dict[str, str]],
+                                     timeout: int = 300,
+                                     configuration: Configuration = None,
+                                     secrets: Secrets = None) -> AWSResponse:
+    """
+    Wait until desired number matches the number of healthy instances
+
+    for each of the auto-scaling groups matching tags provided
+
+    `tags` are  expected as:
+    [{
+        'Key': 'KeyName',
+        'Value': 'KeyValue'
+    },
+    ...
+    ]
+
+    Returns: Integer (number of seconds it took to wait)
+    or sys.maxsize in case of timeout
+    """
+
+    if not tags:
+        raise FailedActivity(
+            "Non-empty tags is required")
+
+    client = aws_client('autoscaling', configuration, secrets)
+
+    start = time.time()
+
+    while True:
+        groups_descr = get_asg_by_tags(tags, client)
+        logger.debug(groups_descr)
+        result = is_desired_equals_healthy(groups_descr)
+
+        if not result:
+            time.sleep(0.1)
+
+            if (time.time() - start) > timeout:
+                result = False
+
+                return maxsize
+        else:
+            return int(time.time() - start)
+
+
+def is_scaling_in_progress(tags: List[Dict[str, str]],
+                           configuration: Configuration = None,
+                           secrets: Secrets = None) -> AWSResponse:
+    """
+    Check if there is any scaling activity in progress for ASG matching tags
+
+    Returns: Boolean
+    """
+
+    if not tags:
+        raise FailedActivity(
+            "Non-empty tags is required")
+
+    client = aws_client('autoscaling', configuration, secrets)
+    groups_descr = get_asg_by_tags(tags, client)
+
+    for group_descr in groups_descr['AutoScalingGroups']:
+        for instance in group_descr['Instances']:
+            if instance['LifecycleState'] != 'InService' \
```

Comment:
Just for my own sanity, the ̀LifecycleState` property is not always set. Notably for On-Demand or Reserved instances. but Maybe this doesn't apply here?

Reply:
I didn't see such cases yet

Reply:
I only came accross it when I created a lone EC2 instances from the AWS dashboard.

Reply:
Let's not worry about it now. We'll see if that comes up.